js/ui/scheduler/ui.scheduler.appointments.js

```diff
@@ -818,6 +823,7 @@ var SchedulerAppointments = CollectionWidget.inherit({
             },
             onDrag: function(args) {
                 correctCoordinates(args.element);
+                adjustDropPosition(args.event, args.element);
```

Comment:
Adjusting drop target anchor by the position of top-left corner of appointment being dragged.